doc/spec/tags.md

```diff
@@ -0,0 +1,171 @@
+
+Note that for compatibility with the first version 2 specification, requests
+to "/v2/<name>/tags/<tag>" will require an accept header with the value of
+"application/vnd.docker.distribution.tag.v1+json" to disambiguate from the
+endpoint "/v2/<name>/tags/list".
+
+### Media Types
+
+Tag's may have the following media types:
+
+| Media Type                                             | Description                                      |
+---------------------------------------------------------|--------------------------------------------------|
+| `application/vnd.docker.distribution.tag.v1+json`      | Base tag object                                  |
+| `application/vnd.docker.distribution.tag.v1+jws`       | tag object wrapped in one or more jws signatures |
+| `application/vnd.docker.distribution.tag.v1+prettyjws` | tag object with jws signatures in pretty format  |
+| `application/vnd.docker.distribution.tags.v1+json`     | List of tag objects.                             |
+
+### Fields
+
+The fields of a tag object are described, as follows:
+
+<dl>
+	<dt>name</dt>
+	<dd>
+		The fully qualified name and tag in docker format, exactly as it would
+		appear on the docker command line. For an image named
+		"example.com/foo/bar" with tag "latest", the value would be
+		"example.com/foo/bar:latest".
+	</dd>
+
+	<dt>description</dt>
+	<dd>A short, human-readable description of the tag and its target.</dd>
+
+	<dt>meta</dt>
+	<dd>Opaque, user-defined fields, keyed by strings. There are no limits to the structure.</dd>
+
+	<dt>target</dt>
+	<dd>
+		<dl>
+			<dt>mediatype</dt>
+			<dd>The media type of the target object.</dd>
+
+			<dt>size</dt>
+			<dd>
+				Size, in bytes, of the targeted content.
+			</dd>
+
+			<dt>digest</dt>
+			<dd>
+				A digest that identifies the target object. The first entry
+				should be considered canonical.
+			</dd>
+		</dl>
+	</dd>
+</dl>
+
+#### Example
+
+The following is an example of a tag object, with media type
+`application/vnd.docker.distribution.tag.v1+json`:
+
+```json
+{
+	"name": "example.com/foo/bar:latest",
+	"description": "A short description about this tag",
+	"meta": {
+		"repo": "github.com/docker/distribution",
+		"commit": "gabcdef",
+	},
+	"target": {
+		"mediatype": "application/vnd.docker.distribution.manifest.v1+json",
+		"digest": "sha256:...",
+		"size": 1024
+	}
+}
+```
+
+If the tag is requested with media type
+`application/vnd.docker.distribution.tag.v1+prettyjws`, signatures for the tag
+will be included with the request. This is similar to how V1, schemaVersion 1
+manifests are currently served. An example follows:
+
+
+```json
+{
+	"name": "example.com/foo/bar:latest",
+	"description": "A short description about this tag",
+	"meta": {
+		"repo": "github.com/docker/distribution",
+		"commit": "gabcdef",
+	},
+	"target": {
+		"mediatype": "application/vnd.docker.distribution.manifest.v1+json",
+		"digest": "sha256:...",
+		"size": 1024
+	},
+	"signatures": ...
```

Comment:
If we're going to embed signatured, let's copy what TUF does and have the top level json contain "signed" and "signatures" objects. It makes it more explicit exactly what was signed and potentially allows you to include unimportant metadata outside the signed section without creating confusion.